### ISSUE TITLE
[Feature Request] Support auto import files that excluded in analysis_options

### ISSUE THREAD

**Issue by lcdsmao** [dart-lang/sdk#46224]
Thank you for taking the time to file an issue!

This tracker is for issues related to:

* Analyzer

If you aren't sure, file the issue here and we'll find the right home for it.
In your issue, please include:

* Dart SDK Version (`dart --version`) 2.13.1
* Whether you are using Windows, MacOSX, or Linux (if applicable) MacOSX
* Whether you are using Chrome, Safari, Firefox, Edge (if applicable)

## Feature request

Usually, we exclude generated files from dart analyzer (e.g. protobuf files).
But we need to use those class/method defined in the generated files.
It is very inconvenient without import auto-complete in this use case.

Excluded:
<img width="500" alt="Screen Shot 2021-06-03 at 9 34 42" src="https://user-images.githubusercontent.com/13031690/120568743-73d91880-c44f-11eb-8177-2f9af2e8c406.png">

Not excluded:
<img width="500" alt="Screen Shot 2021-06-03 at 9 39 07" src="https://user-images.githubusercontent.com/13031690/120568781-8ce1c980-c44f-11eb-8467-56abfc6b11e4.png">

**Comment by devoncarew** [comment]
> Usually, we exclude generated files from dart analyzer (e.g. protobuf files).

Can you talk about why that is? If you're excluding a portion of your application from analysis, you will get some inconsistent behavior in the tooling.

You may try not excluding the generated code and trying another technique for working around whatever issues you see in the generated code (the code failing some lint checks?).

**Comment by lcdsmao** [comment]
@devoncarew 
>(the code failing some lint checks?).

Yes. For example:
- The generated protobuf code failing with:
  - lines_longer_than_80_chars
  - always_use_package_imports
  - constant_identifier_names
- The generated json code failing with:
  -  non_constant_identifier_names

**Comment by lcdsmao** [comment]
Currently, I am using flutter 2.2.0.
I found when I changed to flutter 2.0.6, the import suggestion appears again.
Is this a BUG introduced recently?

**Comment by devoncarew** [comment]
I don't think this is a recent change. The analysis results can change if you open an excluded file - in that situation, we will try and analyze the file.

> Yes. For example:The generated protobuf code failing with

I'd look to ignore the failing lints in a more targeted way - exclude them via an analysis options file at the root of your project or by some other means. Ignoring those files from analysis completely will leave you with inconsistent analysis results and or incomplete info available to tooling.

**Comment by lcdsmao** [comment]
>I'd look to ignore the failing lints in a more targeted way - exclude them via an analysis options file at the root of your project or by some other means. Ignoring those files from analysis completely will leave you with inconsistent analysis results and or incomplete info available to tooling.

I am not sure how to achieve this.
Our `*.pb.dart` files and other dart files stay in the same package.
I think it will be great if we can ignore some rules for some specific files in `analysis_options.yaml`.

**Comment by akihisasengoku** [comment]
I face a similar problem when using [mockito](https://github.com/dart-lang/mockito).

I'm setting `implicit-dynamic: false` in `analysis_options.yaml`.
File generated by `mockito` generated file may include `implicit-dynamic`. therefore, I exclude `*mocks.dart`.
But I am troubled because auto import doesn't work when using generated files.

**Comment by srawlins** [comment]
Sorry this is intentional. Code generators now have more options for creating "clean" code like `// ignore_for_file: type=lint` etc. Files that are excluded from analysis are really excluded from analysis.
